protos/table.proto

```diff
@@ -85,6 +85,7 @@ message Manifest {
   // * 1: deletion files are present
   // * 2: move_stable_row_ids: row IDs are tracked and stable after move operations
   //       (such as compaction), but not updates.
+  // * 8: table metadata are present
```

Comment:
Shouldn't this be 4?

Also, I'm thinking this should be a writer flag only, at least in practice. My use case in my head is that we will use this as configuration for various write operations (how often to auto-cleanup, compact, etc.). But reading the table properly should require interpreting these.

Reply:
Isn't 4 taken by `FLAG_USE_V2_FORMAT_DEPRECATED`?: https://github.com/lancedb/lance/blob/main/rust/lance-table/src/feature_flags.rs#L18

Reply:
Oh okay. Well then we should add that to the proto docs.

Reply:
Done.

Reply:
Could you move it to just be under the `writer_feature_flags`?

Reply:
Ok, I *think* I've done this in [e62bd82](https://github.com/lancedb/lance/pull/2820/commits/e62bd821e532a48bf7ee5026d88e949e5b2efa0b), but let me know if anything else needs editing to make this only apply to writers. I'm not sure if that comment in the proto file needs to change? The code in `feature_flags.rs` + existing comments in `table.proto` makes it seem like the flag should still present be under `reader_feature_flags` even if it only applies to writers?

Reply:
> The code in `feature_flags.rs` + existing comments in `table.proto` makes it seem like the flag should still present be under `reader_feature_flags` even if it only applies to writers?

What do you mean by this? There are separate `can_read_dataset` and `can_write_dataset` methods that check the respective flags.